pkg/domain/infra/abi/containers.go

```diff
@@ -1501,6 +1501,20 @@ func (ic *ContainerEngine) ContainerClone(ctx context.Context, ctrCloneOpts enti
 		return nil, err
 	}
 
+	if ctrCloneOpts.CreateOpts.Pod != "" {
+		allNamespaces := []*specgen.Namespace{&spec.PidNS, &spec.NetNS, &spec.CgroupNS, &spec.IpcNS, &spec.UtsNS}
+		printWarning := false
+		for _, n := range allNamespaces {
+			if !n.IsDefault() {
+				printWarning = true
+			}
+			*n = specgen.Namespace{NSMode: specgen.Default}
+		}
+		if printWarning {
+			logrus.Debug("At least one namespace was set to the default configuration")
+		}
```

Comment:
I don't think this works as expected, I think you should only do the check for the namespaces that are actually shared in the pod instead of all of them.

I tries the following commands as rootless with this patch:
```
podman pod create --name test --share ipc
podman run --network podman  --name con1 alpine ip a
podman container clone --pod test con1
podman start --attach con1-clone
```

In this case since you reset the net namespace we loose the network information but the namespace was never shared so we should not reset it.

Reply:
That looks good but I would like to have a test case for this.

Reply:
test added